aws/rust-runtime/aws-sigv4/src/sign/v4a.rs

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use bytes::{BufMut, BytesMut};
+use num_bigint::BigInt;
+use p256::ecdsa::SigningKey;
+use ring::signature::EcdsaKeyPair;
+use std::io::Write;
+
+const ORDER: &[u32] = &[
+    0xFFFFFFFF, 0x00000000, 0xFFFFFFFF, 0xFFFFFFFF, 0xBCE6FAAD, 0xA7179E84, 0xF3B9CAC2, 0xFC632551,
+];
+const ALGORITHM: &[u8] = b"AWS4-ECDSA-P256-SHA256";
+
+/// Calculates a Sigv4a signature
+pub(crate) fn calculate_signature(signing_key: &EcdsaKeyPair, string_to_sign: &[u8]) -> String {
+    // TODO make this a lazy_static
+    let mut rng = ring::rand::SystemRandom::new();
+    let signature = signing_key.sign(&mut rng, string_to_sign).unwrap();
+    hex::encode(signature.as_ref())
+}
+
+/// Generates a signing key for Sigv4a.
+pub(crate) fn generate_signing_key(access_key: &str, secret_access_key: &str) -> EcdsaKeyPair {
+    // TODO can this be made into a constant?
+    let big_n = BigInt::from_slice(num_bigint::Sign::Plus, ORDER);
+    let big_n_minus_2 = big_n - BigInt::from(2i32);
+    // Capacity is the secret access key length plus the length of "AWS4A"
+    let mut input_key = Vec::with_capacity(secret_access_key.len() + 5);
+    write!(input_key, "AWS4A{secret_access_key}").unwrap();
```

Comment:
Should we be zeroizing these? Looks like the zeroize dependency was never pulled in for V4...